crates/backend/src/codegen.rs

```diff
@@ -971,22 +971,42 @@ impl TryToTokens for ast::ImportFunction {
                 );
             }
             Some(ref ty) => {
-                abi_ret = quote! {
-                    <#ty as wasm_bindgen::convert::FromWasmAbi>::Abi
-                };
-                convert_ret = quote! {
-                    <#ty as wasm_bindgen::convert::FromWasmAbi>
-                        ::from_abi(#ret_ident)
-                };
+                if self.function.r#async {
+                    abi_ret = quote! { js_sys::Promise };
+                    let future = quote! { wasm_bindgen_futures::JsFuture::from(#ret_ident).await };
+                    convert_ret = if self.catch {
```

Comment:
I think for now, would it make more sense to just require `self.catch` to be true? We could figure out later how to rationalize infallible `async` functions, but for now it may be easier to rely on `self.catch` always being true if `self.async` is true`

Reply:
My rationale was that every promise can fail just as every regular JS function can throw. Still, you have the option to declare a regular function as `catch` or `no-catch`.

While failure in JS promises is somewhat more visible than in regular functions, it is fundamentally the same thing. In fact in an JS async function a `try/catch` block will capture both.

I'm not sure what happens if you call a non-catch JS function and it actually throws. I think that it just panics. This could do the same.

But if you prefer not to do that, we can trigger an error on `async && !catch` functions.

Reply:
I think it'll basically make this a bit easier to handle and a bit easier to work with if we require `catch` for now. Exceptions and Rust generally don't play well together right now, unfortunately, and promises give us a good way of "catching" exceptions in a sense.